hkmc2/shared/src/main/scala/hkmc2/bbml/types.scala

```diff
@@ -358,17 +358,26 @@ object Type:
           val u = disjointImpl(a, p.toBasic)(prev)
           val w = disjointImpl(a, q.toBasic)(prev)
           u.flatMap(u => w.map(u ++ _))
-        case (a: InfVar, b: InfVar) if a.uid =/= b.uid => N
+        case (ComposedType(p, q, false), _) =>
+          (disjointImpl(p.toBasic, b)(prev), disjointImpl(q.toBasic, b)(prev)) match
+            case (N, w) => w
+            case (u, N) => u
+            case (S(u), S(w)) => S(u.flatMap(u => w.map(u ++ _)))
+        case (_, ComposedType(p, q, false)) =>
```

Comment:
Intersections on the right can be solved easily, no? Was this case meant to be
```suggestion
        case (_, ComposedType(p, q, true)) =>
```
?

Reply:
Disjointness is symmetric. This case should handle intersection (though it is incorrect).

The `disjoint` function has been rewritten and uses DNF currently.